nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-security/src/main/java/org/apache/nifi/web/security/jwt/JwtAuthenticationFilter.java

```diff
@@ -43,15 +46,27 @@ public Authentication attemptAuthentication(final HttpServletRequest request) {
         // TODO: Refactor request header extraction logic to shared utility as it is duplicated in AccessResource
 
         // get the principal out of the user token
-        final String authorization = request.getHeader(AUTHORIZATION);
+        final String authentication = request.getHeader(AUTHORIZATION);
 
-        // if there is no authorization header, we don't know the user
-        if (authorization == null || !StringUtils.startsWith(authorization, BEARER)) {
+        // if there is no authorization (authentication) header, we don't know the user
+        if (authentication == null || !validJwtFormat(authentication)) {
             return null;
         } else {
             // Extract the Base64 encoded token from the Authorization header
-            final String token = StringUtils.substringAfterLast(authorization, " ");
+            final String token = getTokenFromHeader(authentication);
             return new JwtAuthenticationRequestToken(token, request.getRemoteAddr());
```

Comment:
How do we handle not being able to extract a token from the expected request header?

Reply:
By changing the getTokenFromHeader() method as you suggested above, this can throw an InvalidAuthenticationException that is passed up to NiFiAuthenticationFilter.unsuccessfulAuthentication() which handles this.